### PR TITLE
IZE-776 Goreleaser & Go installer updated

### PR DESCRIPTION
**Pull request by kobrikx**
#### What's new:

 - Goreleaser action updated
 - Settings for Goreleaser updated
 - Go installer action is updated

##### Test configs locally:
Screenshot 2024-08-29 at 16 10 10
Screenshot 2024-08-29 at 16 09 37

##### Github action:
Screenshot 2024-08-29 at 16 17 21
Screenshot 2024-08-29 at 16 17 33
Screenshot 2024-08-29 at 16 17 45

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -82,56 +84,51 @@ archives:
...
-  # Repository to push the app manifest to.
-  bucket:
-    owner: hazelops
-    name: scoop-ize
+# Repository to push the app manifest to.
+scoops:
+  -
+    repository:
+     owner: hazelops
+     name: scoop-ize
```
Comment (AutomationD) [inline .goreleaser.yml:92]: This indent doesn't look right. Owner and name has only 1 indent?

File: .github/workflows/release-dev.build-and-publish.yml [hunk 2]
```diff
@@ -14,9 +14,7 @@ jobs:
 
     steps:
       - name: Install Go
-        uses: actions/setup-go@v2
-        with:
-         go-version: 1.18.x
+        uses: actions/setup-go@v5
```
[the remark this answers is not in the document]
Comment (kobrikx) [reply .github/workflows/release-dev.build-and-publish.yml:17]: No reason. I think it should install latest version.

File: .goreleaser.yml [hunk 3]
```diff
@@ -149,4 +146,4 @@ scoop:
   # GOAMD64 to specify which amd64 version to use if there are multiple versions
   # from the build section.
   # Default is v1.
-#  goamd64: v3
+ # goamd64: v3
```
[the remark this answers is not in the document]
Comment (kobrikx) [reply .goreleaser.yml:149]: I will remove it then.

File: .github/workflows/release-dev.build-and-publish.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kobrikx) [reply .github/workflows/release-dev.build-and-publish.yml:17]: @AutomationD Changed to 1.20.x

[the remark this answers is not in the document]
Comment (AutomationD) [reply .github/workflows/release-dev.build-and-publish.yml:17]: Let's use the latest version but lock it down to 1.23 so there are less chance of breaking.

**Review by AutomationD — APPROVED** [review]

**Pull request merged**